### ISSUE TITLE
docusaurus start and docusaurus build && docusaurus serve have different routing rule  about static folder

### ISSUE THREAD

**Issue by tkow** [facebook/docusaurus#5633]
### Prerequisites

- [x] I'm using the latest version of Docusaurus.
- [x] I have tried the `npm run clear` or `yarn clear` command.
- [x] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [x] I have tried creating a repro with https://new.docusaurus.io
- [x] I have read the console error message carefully (if applicable)

### Description

If you use `docusaurus start` static folder routings are

- no accessable if file dir has "." string and file destination url can  omit /index path
- accessable even if dir name has "." string and the filename is not index.html (can't omit part of url derive from filename)

If you use `docusaurus build and serve` static folder routings are

- no accessable if file name has "." string
- accessable even if dir name has "." string 

As I thought it, Docusaurus  shouldn't rewrite url about static folder resources because its files are tended to be linked by idempotent routing rule and may not depend on hosting server rewriting rule as default.

### Have you read the [Contributing Guidelines on issues]

yes

## Steps to reproduce

To reproduce, we need build step so I made repository for this because I can't find the way to do using sanbox.

Demo: [https://github.com/tkow/docusaurus-url-denied-demo](https://github.com/tkow/docusaurus-url-denied-demo)

1. Pull code from master branch and setup project: `npm install` 
2. Run `npm run start` and access `localhost:3000/html` in browser
3. List of links appears having message of whether or not be accessible when using `docusaurus build and docusaurus serve` and check mismatching between the results of navigating and showing messages by clicking links.
4. Run `npm run build && npm run serve` and access `localhost:3000/html` in browser
5. List of links appears having message of whether or not be accessible when using `docusaurus build and docusaurus serve` and check matching between the results of navigating and showing messages by clicking links.

## Expected behavior

We have same routings between `docsaurus start` and `docsaurus build and docsaurus serve`.
And static route should not be applied rewrite rules as default.

## Actual behavior

We have different routings result between `docsaurus start` and `docsaurus build and docsaurus serve`.

## Your environment

- Public source code: https://github.com/tkow/docusaurus-url-denied-demo
- Public site URL: None
- Docusaurus version used: 
```txt
    "@docusaurus/core": "2.0.0-beta.6"
    "@docusaurus/preset-classic": "2.0.0-beta.6"
```
- Environment name and version (e.g. Chrome 78.0.3904.108, Node.js 10.17.0):  Chrome version: 94.0.4606.61（Official Build） （x86_64, Node.js v12.22.1
- Operating system and version (e.g. Ubuntu 20.04.2 LTS):  macOS Big Sur version 11.5.2（20G95）

## Reproducible demo

https://github.com/tkow/docusaurus-url-denied-demo

**Comment by slorber** [comment]
This issue is very hard to understand for me, can you explain it more simply?

Instead of written text, just give me an example: which request is routed to which static file with start/prod mode. A matrix with factual data would be way easier to understand for me.

Is this issue a duplicate of https://github.com/facebook/docusaurus/issues/4971? 

Dev is using `webpack-dev-server` package
Prod is using `serve` package

The way those servers behave may be slightly different and we may not have options to make them behave 100% in the same way, even if it would be ideal.

**Comment by tkow** [comment]
I took a movies.

These demo structure is located [here](https://github.com/tkow/docusaurus-url-denied-demo/tree/master/static/html).

Dev mode
https://user-images.githubusercontent.com/7004725/135642817-98075bbb-2e5d-4292-a85a-2840a5739ac7.mov
Prod mode
https://user-images.githubusercontent.com/7004725/135643188-38f1102f-8712-45e2-adeb-ebf2f97e600a.mov

Especially we have trouble by disallowing directory name have "." string in production because we want to host our static html integrated by auto-generated by sphinx has  including "." directory. Is the way to avoid this misrouting  with  directory "." included?

**Comment by tkow** [comment]
In additional, when we use
```shell
python -m http.server 8000 --directory ./static 
```

we get different result from above because it doesn't rewrite url and we usually prefer as default hosting static site. Ii is convenient if we switch the static directory configuration that disable rewriting for example.

**Comment by tkow** [comment]
@slorber

I read serve command's implements and I got that serve command is not necessary when we use build artifact as simple static site without backend router. So I found this issue is trivial when use alternative web server or hosting static site storage instead of using `docusaurus serve` command. We simply run `python -m http.server --directory build`  and work well. It hosts theme. So I write adhoc http server script in node.js for my project. Thank you for your opinions.

**Comment by slorber** [comment]
Hi @tkow 

`docusaurus serve` is mostly a tool we provide for convenient local testing. In production you'd rather use a more advanced HTTP server for static files that you can configure according to your own needs, preferably a CDN for performance reasons.

I still think that we can keep this issue open because we should try to make the dev & build/serve similar if possible, but we may close later this issue as wontfix if it's not possible to do so

**Comment by Josh-Cena** [comment]
Closing as wontfix, because we neither control the routing of `start` nor `serve`. I also believe #5420 would have fixed it already.
